From 8b8907b331de6329962a7e4d9101abb934ad90df Mon Sep 17 00:00:00 2001
From: Iceows <mounierr07@gmail.com>
Date: Sun, 8 Oct 2023 22:57:14 +0200
Subject: [PATCH 11/13] treble: add dynamic securize , and some extra 
 functions

- Add Dynamic securize
- Add Autorun
- Add NoLog
- Add Restart sysui
- Add Dump Logs to sdcard
- Add no stock apps (vendor)
---
 base.mk             |  1 +
 phh-on-data.sh      |  6 ++++
 phh-prop-handler.sh | 71 +++++++++++++++++++++++++++++++++++++++++++++
 secure.sh           | 16 ++++++++++
 vndk.rc             | 15 ++++++++++
 5 files changed, 109 insertions(+)
 create mode 100644 secure.sh

diff --git a/base.mk b/base.mk
index a096d0b..18c97d2 100644
--- a/base.mk
+++ b/base.mk
@@ -76,6 +76,7 @@ PRODUCT_PACKAGES += \
         NfcNci \
 
 PRODUCT_COPY_FILES += \
+	device/phh/treble/secure.sh:system/phh/secure.sh \
 	device/phh/treble/rw-system.sh:system/bin/rw-system.sh \
 	device/phh/treble/phh-on-data.sh:system/bin/phh-on-data.sh \
 	device/phh/treble/phh-prop-handler.sh:system/bin/phh-prop-handler.sh \
diff --git a/phh-on-data.sh b/phh-on-data.sh
index 472d16e..1c8b887 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -13,6 +13,12 @@ if getprop persist.sys.phh.caf.media_profile |grep -q true;then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
+if getprop persist.sys.phh.no_stock_apps |grep -q true; then
+	for part in odm vendor; do
+		mount /mnt/phh/empty_dir/ /$part/overlay
+		mount /mnt/phh/empty_dir/ /$part/app
+	done
+fi
 
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 7637cc9..3719902 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -288,3 +288,74 @@ if [ "$1" == "persist.sys.phh.monosim" ]; then
     fi
     exit
 fi
+
+
+#securize
+if [ "$1" == "persist.sys.phh.securize" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/secure ]; then
+        mkdir -p /data/adb/phh
+        cp /system/phh/secure.sh /data/adb/phh/secure
+    fi
+    /system/bin/sh /data/adb/phh/secure
+    exit
+fi
+
+#autorun
+if [ "$1" == "persist.sys.phh.autorun" ]; then
+    if [[ "$prop_value" != "true" && "$prop_value" != "false" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/run ]; then
+        mkdir -p /data/adb/phh
+        touch /data/adb/phh/run
+    fi
+    /system/bin/sh /data/adb/phh/run
+    exit
+fi
+
+#nolog
+if [ "$1" == "persist.sys.phh.nolog" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        setprop ctl.stop logd
+        setprop ctl.stop traced
+        setprop ctl.stop traced_probes
+    else
+        setprop ctl.start traced_probes
+        setprop ctl.start traced
+        setprop ctl.start logd
+    fi
+    exit
+fi
+
+#restart_sysui
+if [ "$1" == "sys.phh.restart_sysui" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        pkill systemui
+    fi
+    exit
+fi
+
+#dump_logs
+if [ "$1" == "sys.phh.dump_logs" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        logcat -b all -d > /sdcard/logs.txt
+    fi
+    exit
+fi
diff --git a/secure.sh b/secure.sh
new file mode 100644
index 0000000..d776832
--- /dev/null
+++ b/secure.sh
@@ -0,0 +1,16 @@
+(getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
+done
+resetprop_phh ro.build.tags release-keys
+resetprop_phh ro.boot.vbmeta.device_state locked
+resetprop_phh ro.boot.verifiedbootstate green
+resetprop_phh ro.boot.flash.locked 1
+resetprop_phh ro.boot.veritymode enforcing
+resetprop_phh ro.boot.warranty_bit 0
+resetprop_phh ro.warranty_bit 0
+resetprop_phh ro.debuggable 0
+resetprop_phh ro.secure 1
+resetprop_phh ro.build.type user
+resetprop_phh ro.build.selinux 0
+resetprop_phh ro.adb.secure 1
+setprop ctl.restart adbd
\ No newline at end of file
diff --git a/vndk.rc b/vndk.rc
index 470b9ac..5d00b88 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -67,6 +67,21 @@ on property:ro.sf.lcd_density=*
 on property:persist.bluetooth.system_audio_hal.enabled=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.bluetooth.system_audio_hal.enabled"
 
+on property:persist.sys.phh.securize=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.securize"
+
+on property:persist.sys.phh.autorun=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.autorun"
+
+on property:persist.sys.phh.nolog=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.nolog"
+
+on property:sys.phh.restart_sysui=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.restart_sysui"
+
+on property:sys.phh.dump_logs=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.dump_logs"
+
 on property:init.svc.ril-proxy=stopped && property:persist.sys.phh.restart_ril=true
     start ril-proxy
 
-- 
2.33.0.windows.2

